CookieMediator/src/main/java/org/wso2/carbon/connector/creatiocrm/headers/BPMCSRFCookie.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.synapse.MessageContext;
+
+public class BPMCSRFCookie extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(BPMCSRFCookie.class);
+    private static String COOKIE_NAME = "set-cookie";
+    private static String COOKIE_NAME_CAMELCASE = "Set-Cookie";
+    private static String COOKIE_BPMCSRF = "BPMCSRF=";
+    private static String AUTH_COMPLETE_PROPERTY = "authCompleteCookie";
+    private static String BPMSESSIONID = "BPMSESSIONID";
+    private static String USERNAME = "UserName";
+    private static String BPMLOADER = "BPMLOADER";
+    private static String ASPXAUTH = ".ASPXAUTH";
+    private static String BPMCSRF = "BPMCSRF";
+
+    public boolean mediate(MessageContext context) {
+
+        String allCookies = "";
+        Map<String, String> cookiesMap = new HashMap<String, String>();
+        Object headers = ((Axis2MessageContext) context)
+                .getAxis2MessageContext().getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+        Map<String, String> headerMap = (Map) headers;
+
+        if (headerMap.containsKey(COOKIE_NAME_CAMELCASE) || headerMap.containsKey(COOKIE_NAME)) {
+            String cookie = headerMap.get(COOKIE_NAME_CAMELCASE);
+            if(cookie.isEmpty()){
+                cookie = headerMap.get(COOKIE_NAME);
+            }
+            String authCookiesVal = cookie.split(";")[0];
+            if (cookie.contains(COOKIE_BPMCSRF)) {
+                context.setProperty(BPMCSRF, authCookiesVal.split(":")[1]);
+            }
+            cookiesMap.put(authCookiesVal.split("=")[0], cookie.split(";")[0]);
+        }
+
+        Map<String, List<String>> excessHeadersMap = (Map) ((Axis2MessageContext) context).getAxis2MessageContext().getProperty("EXCESS_TRANSPORT_HEADERS");
+        if (excessHeadersMap.containsKey(COOKIE_NAME_CAMELCASE) || excessHeadersMap.containsKey(COOKIE_NAME)) {
+            List<String> cookies = excessHeadersMap.get(COOKIE_NAME_CAMELCASE);
+            if (cookies == null) {
+                cookies = excessHeadersMap.get(COOKIE_NAME);
+            }
+            if (cookies != null) {
+                for(String cookie : cookies) {
+                    String authCookiesVal = cookie.split(";")[0];
+                    cookiesMap.put(authCookiesVal.split("=")[0], cookie.split(";")[0]);
+                    if (cookie.contains(COOKIE_BPMCSRF)) {
+                        context.setProperty(BPMCSRF, authCookiesVal.split("=")[1]);
+                    }
+                }
+            }
+        }
+        allCookies = getCookieVal(cookiesMap, BPMSESSIONID, allCookies);
+        allCookies = getCookieVal(cookiesMap, USERNAME, allCookies);
+        allCookies = getCookieVal(cookiesMap, BPMLOADER, allCookies);
+        allCookies = getCookieVal(cookiesMap, ASPXAUTH, allCookies);
+        allCookies = getCookieVal(cookiesMap, BPMCSRF, allCookies);
+        context.setProperty(AUTH_COMPLETE_PROPERTY, allCookies);
```

Comment:
shall we move this to a different function and do the logic in one go, repeating the logic for each iteration is not needed ?

Reply:
Also we can breakdown and have meaningful names explaining the functionality each does.

Reply:
Updated the code accordingly